### PR TITLE
Enable MDNorm to remove background

### LINKED ISSUE #30926: Efficient background removal in MD
### Expected behavior

This is SNS spectroscopy [story 59](https://code.ornl.gov/sns-hfir-scse/spectroscopy/spectroscopy/-/issues/59)

It is expected to implement an approach that would be to create the MDEvent workspace for the background in Q_lab, meaning a single orientation, and use a new algorithm, similar to MDNorm to rotate and histogram this workspace in one step.

```math
Q_{lab}=RQ_{sample}
```

where $`R`$ is the goniometer matrix.

The MDNorm algorithm does two things with the input workspace. 1. It bins the data on the desired grid. 2. It calculates the normalization on that grid. The binning occurs in the [binInputWorkspace](https://github.com/mantidproject/mantid/blob/8558780bf8f01a6ba3866f114a7fc959822fa2bf/Framework/MDAlgorithms/src/MDNorm.cpp#L921-L1055) function. Everything after bin validation (line 930) and up to line 1032 needs to be repeated for as many times as I have ExperimentInfo objects in the data workspace, with changing the Qtransform matrix after line 949 into R\*Qtransform. For normalization is even easier. In the [calculateNormalization](https://github.com/mantidproject/mantid/blob/8558780bf8f01a6ba3866f114a7fc959822fa2bf/Framework/MDAlgorithms/src/MDNorm.cpp#L1146) function just read the goniometer setting from the data workspace experiment info instead of the background experiment info. The proton charge is from the background. So `currentExptInfo` on line 1149 refers to the data, but then on line 1169 it should change to point to experiment info number 0 in the background. Note that this already done in a loop on https://github.com/mantidproject/mantid/blob/8558780bf8f01a6ba3866f114a7fc959822fa2bf/Framework/MDAlgorithms/src/MDNorm.cpp#L480, but one needs to loop over the experiment infos in the data, not in the background. The `outputDataWS` will have a single experiment info.

The script with the current version and a commented section for the new version is attached: [efficient_background.py](/sns-hfir-scse/spectroscopy/spectroscopy/uploads/b459faf7f810ca054a8577f1eee9470d/efficient_background.py) Note that the data used is from mantid systemtests.

### Actual behavior

Currently, when I measure the angle independent background, I take the event workspace for the background, treat it the same as the data, set the goniometer to the same value as the data, convert to MD, then proceed with the processing the same as for the data. The disadvantage of this approach is that for multiple rotations I create a huge MDEvent workspace for the background, sometimes an order of magnitude larger than the data. This becomes problematic if I have many rotations. The only difference, since I copy the background events as many times as I have runs in the real data. See for example https://github.com/AndreiSavici/DGS_SC_scripts/blob/master/reduce_data_to_MDE.py#L455-L471

### PR DESCRIPTION
**Pull request by wdzhou**
**Description of work.**

This fixes #30926.

Fixes #xxxx. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/framework/MDNormBackgroundHYSPECTest.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+        # prepare sample MD
+        self.prepare_md(input_ws_name='sum', merged_md_name='merged', min_log_value=10, max_log_value=12)
+        # Prepare background workspace
+        # old way - use reduced_1 as the background
+        self.prepare_background(input_md='reduced_1', reference_sample_mde='merged',
+                                background_md='background_MDE')
+
+        # do MDNorm to sample data
+        MDNorm(InputWorkspace='merged',
```
Comment (wdzhou) [inline Testing/SystemTests/tests/framework/MDNormBackgroundHYSPECTest.py:122]: @jrmcgaha-dev  Could you run @AndreiSavici 's script to generate output and compare with this?  The modification to Andrie's shall be 
1. Load only run 13655
2. Minimum log value in GenerateEventFilters shall be set to 10, which is not specified in his script
3. Maximum log value in GenerateEventFilters shall be set to 12 from 90.

Comment (jrmcgaha-dev) [reply Testing/SystemTests/tests/framework/MDNormBackgroundHYSPECTest.py:122]: 1. Correction: 13656

File: Testing/SystemTests/tests/framework/MDNormBackgroundHYSPECTest.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+               Dimension0Name='QDimension1',
+               Dimension0Binning='-5,0.05,5',
+               Dimension1Name='QDimension2',
+               Dimension1Binning='-5,0.05,5',
+               Dimension2Name='DeltaE',
+               Dimension2Binning='-2,2',
+               Dimension3Name='QDimension0',
+               Dimension3Binning='-0.5,0.5',
+               SymmetryOperations='x,y,z;x,-y,z;x,y,-z;x,-y,-z',
+               TemporaryDataWorkspace='backgroundMDH',
```
Comment (wdzhou) [inline Testing/SystemTests/tests/framework/MDNormBackgroundHYSPECTest.py:237]: @jrmcgaha-dev I made a mistake here.

**Comment by wdzhou** [comment]
Replaced by https://github.com/mantidproject/mantid/pull/31058

**Pull request closed without merging**